presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnection.java

```diff
@@ -634,7 +634,7 @@ String getStartTransactionSql()
     StatementClient startQuery(String sql, Map<String, String> sessionPropertiesOverride)
     {
         String source = firstNonNull(clientInfo.get("ApplicationName"), "presto-jdbc");
-        String traceToken = firstNonNull(clientInfo.get("TraceToken"), generateTraceToken());
+        String traceToken = clientInfo.get("TraceToken");
```

Comment:
Change this variable to be an `Optional<String>`. That provides better null safety by avoiding a local variable which can contain null.